### PR TITLE
registry(engine): Simplify worker registration with generics

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?

Issue Number: ref #5805

### What is changed and how it works?
- Made `NewSimpleWorkerFactory` generic.

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No.

##### Do you need to update user documentation, design documentation or monitoring documentation?
No.

### Release note 

```release-note
None.
```

### CODE REVIEW

File: engine/lib/registry/factory.go [hunk 1]
```diff
@@ -40,35 +41,45 @@ type WorkerFactory interface {
...
 	}
 }
 
 // NewWorkerImpl implements WorkerFactory.NewWorkerImpl
-func (f *SimpleWorkerFactory) NewWorkerImpl(
+func (f *SimpleWorkerFactory[T, C]) NewWorkerImpl(
 	ctx *dcontext.Context,
 	workerID libModel.WorkerID,
 	masterID libModel.MasterID,
 	config WorkerConfig,
```
Comment (lance6716) [inline engine/lib/registry/factory.go:74]: can we require config is of type C?

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: engine/lib/registry/factory.go (hunk 1 above)
Comment (liuzix) [reply engine/lib/registry/factory.go:74]: It seems you cannot implement the `WorkerFactory` interface this way. There has to be a place where you do a `type erasure`.

**Review by lance6716 — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**